genesis/src/main.rs

```diff
@@ -30,10 +31,14 @@ pub enum AccountFileFormat {
     Keypair,
 }
 
-fn pubkey_from_file(key_file: &str) -> Result<Pubkey, Box<dyn error::Error>> {
-    read_pubkey_file(key_file)
-        .or_else(|_| read_keypair_file(key_file).map(|keypair| keypair.pubkey()))
-        .map_err(|err| format!("Failed to read {}: {}", key_file, err).into())
+fn required_pubkey(matches: &ArgMatches<'_>, name: &str) -> Result<Pubkey, Box<dyn error::Error>> {
+    pubkey_of(matches, name).map(Ok).unwrap_or_else(|| {
+        Err(format!(
+            "Invalid pubkey or file: {}",
+            matches.value_of(name).unwrap_or("<no arg>")
```

Comment:
Since you're only using this method on args are `required` by clap, you don't need this empty case handling. `unwrap()` will do

Reply:
Yeah! That reasoning is correct!  58947fac2bcf552ccb3a576be63ac7df94093920